File: pureMood_backend/models/ChatMessage.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const ChatMessage = sequelize.define('ChatMessage', {
  message_id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  session_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'chat_sessions',
      key: 'session_id'
    },
    onDelete: 'CASCADE'
  },
  role: {
    type: DataTypes.ENUM('user', 'assistant', 'system'),
    allowNull: false
  },
  content: {
    type: DataTypes.TEXT,
    allowNull: false
  },
  safety_flags: {
    type: DataTypes.JSON,
    allowNull: true,
    defaultValue: [],
    comment: 'Array of safety warnings if detected'
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'chat_messages',
  timestamps: false
});

module.exports = ChatMessage;

File: puremood_frontend/lib/models/chat_message.dart
class ChatMessage {
  final int? messageId;
  final String role; // 'user' or 'assistant'
  final String content;
  final List<String> safetyFlags;
  final DateTime? createdAt;

  ChatMessage({
    this.messageId,
    required this.role,
    required this.content,
    this.safetyFlags = const [],
    this.createdAt,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      messageId: json['message_id'],
      role: json['role'] ?? 'assistant',
      content: json['content'] ?? '',
      safetyFlags: json['safety_flags'] != null 
        ? List<String>.from(json['safety_flags']) 
        : [],
      createdAt: json['created_at'] != null 
        ? DateTime.parse(json['created_at']) 
        : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'role': role,
      'content': content,
    };
  }

  bool get isUser => role == 'user';
  bool get isAssistant => role == 'assistant';
  bool get hasSafetyFlags => safetyFlags.isNotEmpty;
}
